lib/services/data/budget/providers/task_budget_total_value_provider.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:seren_ai_flutter/services/data/budget/providers/task_budget_items_stream_provider.dart';
+import 'package:seren_ai_flutter/services/data/tasks/providers/task_by_id_stream_provider.dart';
+import 'package:seren_ai_flutter/services/data/tasks/providers/tasks_by_parent_stream_provider.dart';
+import 'package:seren_ai_flutter/services/data/tasks/providers/tasks_by_project_stream_provider.dart';
+
+/// Returns the total value of the task budget items for a task (either a child or a phase)
+final taskBudgetItemsTotalValueProvider =
+    Provider.autoDispose.family<double, String>((ref, taskId) {
+  final task =
+      ref.watch(taskByIdStreamProvider(taskId).select((value) => value.value));
+
+  if (task?.isPhase ?? false) {
+    return ref.watch(_phaseBudgetTotalValueProvider(taskId));
+  } else {
+    return ref.watch(_childTaskBudgetTotalValueProvider(taskId));
+  }
+});
+
+/// Returns the total value of the task budget items for a project
+final projectBudgetTotalValueProvider =
+    Provider.autoDispose.family<double, String>((ref, projectId) {
+  final tasks = ref.watch(tasksByProjectStreamProvider(projectId)).value;
+  if (tasks == null) {
+    return 0.0;
+  }
+
+  // get only child tasks to avoid double counting
+  final childTaskIds =
```

Comment:
we have a lot of derived calculations here from other providers ... but shouldn't we be using separate SQL queries here instead? this allows easier reused logic in the future while also avoiding these more expensive O(n) operations all the time that will not be sustainable once there are 1000s of tasks

Reply:
this is okay for now - but can you please leave a note in the providers you added that you think should be switched to SQL queries either locally or as edge functions? 

Mark them as TODO p3 (Provider Optimization): move to SQL queries once there are 1000s of tasks